pilot/pkg/model/sidecar.go

```diff
@@ -228,6 +232,14 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *Config, configNamespa
 		out.destinationRules[s.Hostname] = ps.DestinationRule(&dummyNode, s)
 	}
 
+	// Assign namespace dependencies
+	out.namespaceDependencies = make(map[string]struct{})
+	for _, egress := range out.EgressListeners {
+		for ns := range egress.listenerHosts {
```

Comment:
so you dont have to go through listener hosts. you go through the services in each egress